#transfer all data from uppmax
# Scp xung0001@rackham.uppmax.uu.se:/crex/proj/snic2021-23-14/Xue/tissue_RNA_seq/single/kallisto/SRR*/*.tsv /Users/xung0001/Desktop/
# load all overlapping s_as pairs
co_non <- read.table("/Users/xung0001/Desktop/coding-noncoding.txt",head=T,sep = ",")
non_non <- read.table("/Users/xung0001/Desktop/noncoding_noncoding.txt",head=T,sep = ",")
co_coding <- read.table("/Users/xung0001/Desktop/coding_anticoding.txt",head=T,sep = ",")
# unite together and define as s_as table
s_as <- rbing(co_coding,co_non)
s_as <- rbind(s_as,non_non)
# loading expression profile generated by from Kallisto method
root1<- read.csv("/Users/xung0001/root1.tsv",sep="\t")
root1 <- as.data.frame(root1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(root1$target_id)==11, "yes","no")
root1 <- root1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(root1$target_id,nchar(root1$target_id),nchar(root1$target_id))=="1")
root1 <- root1[x,]
# remove the .1 from transcript id
root1$target_id <- substr(root1$target_id, 1, nchar(root1$target_id)-2)
row.names(root1) <- root1$target_id 
#find the expression value from root1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sroot1exp <- rep(NA,dim(s_as)[1])
s_as$asroot1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% root1$target_id){
   exps<- paste0(root1[s_as[i,1],]$tpm," ",exps )}
   if(length(exps>0)){s_as$sroot1exp[i] <- exps}
  if(s_as[i,2]%in% root1$target_id){
    expas<- paste0(root1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asroot1exp[i] <- expas}
  }
# loading expression profile generated by from Kallisto method
root2<- read.csv("/Users/xung0001/Desktop/root2.tsv",sep="\t")
root2 <- as.data.frame(root2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(root2$target_id)==11, "yes","no")
root2 <- root2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(root2$target_id,nchar(root2$target_id),nchar(root2$target_id))=="1")
root2 <- root2[x,]
# remove the .1 from transcript id
root2$target_id <- substr(root2$target_id, 1, nchar(root2$target_id)-2)
row.names(root2) <- root2$target_id 
#find the expression value from root2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sroot2exp <- rep(NA,dim(s_as)[1])
s_as$asroot2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% root2$target_id){
    exps<- paste0(root2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sroot2exp[i] <- exps}
  if(s_as[i,2]%in% root2$target_id){
    expas<- paste0(root2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asroot2exp[i] <- expas}
}

root3<- read.csv("/Users/xung0001/Desktop/root3.tsv",sep="\t")
root3 <- as.data.frame(root3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(root3$target_id)==11, "yes","no")
root3 <- root3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(root3$target_id,nchar(root3$target_id),nchar(root3$target_id))=="1")
root3 <- root3[x,]
# remove the .1 from transcript id
root3$target_id <- substr(root3$target_id, 1, nchar(root3$target_id)-2)
row.names(root3) <- root3$target_id 
#find the expression value from root3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sroot3exp <- rep(NA,dim(s_as)[1])
s_as$asroot3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% root3$target_id){
    exps<- paste0(root3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sroot3exp[i] <- exps}
  if(s_as[i,2]%in% root3$target_id){
    expas<- paste0(root3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asroot3exp[i] <- expas}
}

# shoot1
# loading expression profile generated by from Kallisto method
shoot1<- read.csv("/Users/xung0001/Desktop/shoot1.tsv",sep="\t")
shoot1 <- as.data.frame(shoot1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(shoot1$target_id)==11, "yes","no")
shoot1 <- shoot1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(shoot1$target_id,nchar(shoot1$target_id),nchar(shoot1$target_id))=="1")
shoot1 <- shoot1[x,]
# remove the .1 from transcript id
shoot1$target_id <- substr(shoot1$target_id, 1, nchar(shoot1$target_id)-2)
row.names(shoot1) <- shoot1$target_id 
#find the expression value from shoot1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sshoot1exp <- rep(NA,dim(s_as)[1])
s_as$asshoot1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% shoot1$target_id){
    exps<- paste0(shoot1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sshoot1exp[i] <- exps}
  if(s_as[i,2]%in% shoot1$target_id){
    expas<- paste0(shoot1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asshoot1exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
shoot2<- read.csv("/Users/xung0001/Desktop/shoot2.tsv",sep="\t")
shoot2 <- as.data.frame(shoot2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(shoot2$target_id)==11, "yes","no")
shoot2 <- shoot2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(shoot2$target_id,nchar(shoot2$target_id),nchar(shoot2$target_id))=="1")
shoot2 <- shoot2[x,]
# remove the .1 from transcript id
shoot2$target_id <- substr(shoot2$target_id, 1, nchar(shoot2$target_id)-2)
row.names(shoot2) <- shoot2$target_id 
#find the expression value from shoot2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sshoot2exp <- rep(NA,dim(s_as)[1])
s_as$asshoot2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% shoot2$target_id){
    exps<- paste0(shoot2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sshoot2exp[i] <- exps}
  if(s_as[i,2]%in% shoot2$target_id){
    expas<- paste0(shoot2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asshoot2exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
shoot3<- read.csv("/Users/xung0001/Desktop/shoot3.tsv",sep="\t")
shoot3 <- as.data.frame(shoot3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(shoot3$target_id)==11, "yes","no")
shoot3 <- shoot3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(shoot3$target_id,nchar(shoot3$target_id),nchar(shoot3$target_id))=="1")
shoot3 <- shoot3[x,]
# remove the .1 from transcript id
shoot3$target_id <- substr(shoot3$target_id, 1, nchar(shoot3$target_id)-2)
row.names(shoot3) <- shoot3$target_id 
#find the expression value from shoot3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sshoot3exp <- rep(NA,dim(s_as)[1])
s_as$asshoot3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% shoot3$target_id){
    exps<- paste0(shoot3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sshoot3exp[i] <- exps}
  if(s_as[i,2]%in% shoot3$target_id){
    expas<- paste0(shoot3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asshoot3exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
infloresence1<- read.csv("/Users/xung0001/Desktop/infloresence1.tsv",sep="\t")
infloresence1 <- as.data.frame(infloresence1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(infloresence1$target_id)==11, "yes","no")
infloresence1 <- infloresence1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(infloresence1$target_id,nchar(infloresence1$target_id),nchar(infloresence1$target_id))=="1")
infloresence1 <- infloresence1[x,]
# remove the .1 from transcript id
infloresence1$target_id <- substr(infloresence1$target_id, 1, nchar(infloresence1$target_id)-2)
row.names(infloresence1) <- infloresence1$target_id 
#find the expression value from infloresence1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sinfloresence1exp <- rep(NA,dim(s_as)[1])
s_as$asinfloresence1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% infloresence1$target_id){
    exps<- paste0(infloresence1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sinfloresence1exp[i] <- exps}
  if(s_as[i,2]%in% infloresence1$target_id){
    expas<- paste0(infloresence1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asinfloresence1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
infloresence2<- read.csv("/Users/xung0001/Desktop/infloresence2.tsv",sep="\t")
infloresence2 <- as.data.frame(infloresence2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(infloresence2$target_id)==11, "yes","no")
infloresence2 <- infloresence2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(infloresence2$target_id,nchar(infloresence2$target_id),nchar(infloresence2$target_id))=="1")
infloresence2 <- infloresence2[x,]
# remove the .1 from transcript id
infloresence2$target_id <- substr(infloresence2$target_id, 1, nchar(infloresence2$target_id)-2)
row.names(infloresence2) <- infloresence2$target_id 
#find the expression value from infloresence2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sinfloresence2exp <- rep(NA,dim(s_as)[1])
s_as$asinfloresence2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% infloresence2$target_id){
    exps<- paste0(infloresence2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sinfloresence2exp[i] <- exps}
  if(s_as[i,2]%in% infloresence2$target_id){
    expas<- paste0(infloresence2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asinfloresence2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
infloresence3<- read.csv("/Users/xung0001/Desktop/infloresence3.tsv",sep="\t")
infloresence3 <- as.data.frame(infloresence3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(infloresence3$target_id)==11, "yes","no")
infloresence3 <- infloresence3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(infloresence3$target_id,nchar(infloresence3$target_id),nchar(infloresence3$target_id))=="1")
infloresence3 <- infloresence3[x,]
# remove the .1 from transcript id
infloresence3$target_id <- substr(infloresence3$target_id, 1, nchar(infloresence3$target_id)-2)
row.names(infloresence3) <- infloresence3$target_id 
#find the expression value from infloresence3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sinfloresence3exp <- rep(NA,dim(s_as)[1])
s_as$asinfloresence3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% infloresence3$target_id){
    exps<- paste0(infloresence3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sinfloresence3exp[i] <- exps}
  if(s_as[i,2]%in% infloresence3$target_id){
    expas<- paste0(infloresence3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asinfloresence3exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
silique1<- read.csv("/Users/xung0001/Desktop/silique1.tsv",sep="\t")
silique1 <- as.data.frame(silique1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(silique1$target_id)==11, "yes","no")
silique1 <- silique1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(silique1$target_id,nchar(silique1$target_id),nchar(silique1$target_id))=="1")
silique1 <- silique1[x,]
# remove the .1 from transcript id
silique1$target_id <- substr(silique1$target_id, 1, nchar(silique1$target_id)-2)
row.names(silique1) <- silique1$target_id 
#find the expression value from silique1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$ssilique1exp <- rep(NA,dim(s_as)[1])
s_as$assilique1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% silique1$target_id){
    exps<- paste0(silique1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$ssilique1exp[i] <- exps}
  if(s_as[i,2]%in% silique1$target_id){
    expas<- paste0(silique1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$assilique1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
silique2<- read.csv("/Users/xung0001/Desktop/silique2.tsv",sep="\t")
silique2 <- as.data.frame(silique2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(silique2$target_id)==11, "yes","no")
silique2 <- silique2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(silique2$target_id,nchar(silique2$target_id),nchar(silique2$target_id))=="1")
silique2 <- silique2[x,]
# remove the .1 from transcript id
silique2$target_id <- substr(silique2$target_id, 1, nchar(silique2$target_id)-2)
row.names(silique2) <- silique2$target_id 
#find the expression value from silique2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$ssilique2exp <- rep(NA,dim(s_as)[1])
s_as$assilique2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% silique2$target_id){
    exps<- paste0(silique2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$ssilique2exp[i] <- exps}
  if(s_as[i,2]%in% silique2$target_id){
    expas<- paste0(silique2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$assilique2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
silique3<- read.csv("/Users/xung0001/Desktop/silique3.tsv",sep="\t")
silique3 <- as.data.frame(silique3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(silique3$target_id)==11, "yes","no")
silique3 <- silique3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(silique3$target_id,nchar(silique3$target_id),nchar(silique3$target_id))=="1")
silique3 <- silique3[x,]
# remove the .1 from transcript id
silique3$target_id <- substr(silique3$target_id, 1, nchar(silique3$target_id)-2)
row.names(silique3) <- silique3$target_id 
#find the expression value from silique3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$ssilique3exp <- rep(NA,dim(s_as)[1])
s_as$assilique3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% silique3$target_id){
    exps<- paste0(silique3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$ssilique3exp[i] <- exps}
  if(s_as[i,2]%in% silique3$target_id){
    expas<- paste0(silique3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$assilique3exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
hypocotyl1<- read.csv("/Users/xung0001/Desktop/hypocotyl1.tsv",sep="\t")
hypocotyl1 <- as.data.frame(hypocotyl1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(hypocotyl1$target_id)==11, "yes","no")
hypocotyl1 <- hypocotyl1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(hypocotyl1$target_id,nchar(hypocotyl1$target_id),nchar(hypocotyl1$target_id))=="1")
hypocotyl1 <- hypocotyl1[x,]
# remove the .1 from transcript id
hypocotyl1$target_id <- substr(hypocotyl1$target_id, 1, nchar(hypocotyl1$target_id)-2)
row.names(hypocotyl1) <- hypocotyl1$target_id 
#find the expression value from hypocotyl1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$shypocotyl1exp <- rep(NA,dim(s_as)[1])
s_as$ashypocotyl1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% hypocotyl1$target_id){
    exps<- paste0(hypocotyl1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$shypocotyl1exp[i] <- exps}
  if(s_as[i,2]%in% hypocotyl1$target_id){
    expas<- paste0(hypocotyl1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ashypocotyl1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
hypocotyl2<- read.csv("/Users/xung0001/Desktop/hypocotyl2.tsv",sep="\t")
hypocotyl2 <- as.data.frame(hypocotyl2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(hypocotyl2$target_id)==11, "yes","no")
hypocotyl2 <- hypocotyl2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(hypocotyl2$target_id,nchar(hypocotyl2$target_id),nchar(hypocotyl2$target_id))=="1")
hypocotyl2 <- hypocotyl2[x,]
# remove the .1 from transcript id
hypocotyl2$target_id <- substr(hypocotyl2$target_id, 1, nchar(hypocotyl2$target_id)-2)
row.names(hypocotyl2) <- hypocotyl2$target_id 
#find the expression value from hypocotyl2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$shypocotyl2exp <- rep(NA,dim(s_as)[1])
s_as$ashypocotyl2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% hypocotyl2$target_id){
    exps<- paste0(hypocotyl2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$shypocotyl2exp[i] <- exps}
  if(s_as[i,2]%in% hypocotyl2$target_id){
    expas<- paste0(hypocotyl2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ashypocotyl2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
hypocotyl3<- read.csv("/Users/xung0001/Desktop/hypocotyl3.tsv",sep="\t")
hypocotyl3 <- as.data.frame(hypocotyl3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(hypocotyl3$target_id)==11, "yes","no")
hypocotyl3 <- hypocotyl3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(hypocotyl3$target_id,nchar(hypocotyl3$target_id),nchar(hypocotyl3$target_id))=="1")
hypocotyl3 <- hypocotyl3[x,]
# remove the .1 from transcript id
hypocotyl3$target_id <- substr(hypocotyl3$target_id, 1, nchar(hypocotyl3$target_id)-2)
row.names(hypocotyl3) <- hypocotyl3$target_id 
#find the expression value from hypocotyl3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$shypocotyl3exp <- rep(NA,dim(s_as)[1])
s_as$ashypocotyl3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% hypocotyl3$target_id){
    exps<- paste0(hypocotyl3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$shypocotyl3exp[i] <- exps}
  if(s_as[i,2]%in% hypocotyl3$target_id){
    expas<- paste0(hypocotyl3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ashypocotyl3exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
hypocotyl4<- read.csv("/Users/xung0001/Desktop/hypocotyl4.tsv",sep="\t")
hypocotyl4 <- as.data.frame(hypocotyl4)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(hypocotyl4$target_id)==11, "yes","no")
hypocotyl4 <- hypocotyl4[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(hypocotyl4$target_id,nchar(hypocotyl4$target_id),nchar(hypocotyl4$target_id))=="1")
hypocotyl4 <- hypocotyl4[x,]
# remove the .1 from transcript id
hypocotyl4$target_id <- substr(hypocotyl4$target_id, 1, nchar(hypocotyl4$target_id)-2)
row.names(hypocotyl4) <- hypocotyl4$target_id 
#find the expression value from hypocotyl4 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$shypocotyl4exp <- rep(NA,dim(s_as)[1])
s_as$ashypocotyl4exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% hypocotyl4$target_id){
    exps<- paste0(hypocotyl4[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$shypocotyl4exp[i] <- exps}
  if(s_as[i,2]%in% hypocotyl4$target_id){
    expas<- paste0(hypocotyl4[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ashypocotyl4exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
sgerminating1<- read.csv("/Users/xung0001/Desktop/sgerminating1.tsv",sep="\t")
sgerminating1 <- as.data.frame(sgerminating1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(sgerminating1$target_id)==11, "yes","no")
sgerminating1 <- sgerminating1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(sgerminating1$target_id,nchar(sgerminating1$target_id),nchar(sgerminating1$target_id))=="1")
sgerminating1 <- sgerminating1[x,]
# remove the .1 from transcript id
sgerminating1$target_id <- substr(sgerminating1$target_id, 1, nchar(sgerminating1$target_id)-2)
row.names(sgerminating1) <- sgerminating1$target_id 
#find the expression value from sgerminating1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$ssgerminating1exp <- rep(NA,dim(s_as)[1])
s_as$assgerminating1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% sgerminating1$target_id){
    exps<- paste0(sgerminating1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$ssgerminating1exp[i] <- exps}
  if(s_as[i,2]%in% sgerminating1$target_id){
    expas<- paste0(sgerminating1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$assgerminating1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
sgerminating2<- read.csv("/Users/xung0001/Desktop/sgerminating2.tsv",sep="\t")
sgerminating2 <- as.data.frame(sgerminating2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(sgerminating2$target_id)==11, "yes","no")
sgerminating2 <- sgerminating2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(sgerminating2$target_id,nchar(sgerminating2$target_id),nchar(sgerminating2$target_id))=="1")
sgerminating2 <- sgerminating2[x,]
# remove the .1 from transcript id
sgerminating2$target_id <- substr(sgerminating2$target_id, 1, nchar(sgerminating2$target_id)-2)
row.names(sgerminating2) <- sgerminating2$target_id 
#find the expression value from sgerminating2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$ssgerminating2exp <- rep(NA,dim(s_as)[1])
s_as$assgerminating2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% sgerminating2$target_id){
    exps<- paste0(sgerminating2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$ssgerminating2exp[i] <- exps}
  if(s_as[i,2]%in% sgerminating2$target_id){
    expas<- paste0(sgerminating2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$assgerminating2exp[i] <- expas}
}

# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(ovule1$target_id)==11, "yes","no")
ovule1 <- ovule1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(ovule1$target_id,nchar(ovule1$target_id),nchar(ovule1$target_id))=="1")
ovule1 <- ovule1[x,]
# remove the .1 from transcript id
ovule1$target_id <- substr(ovule1$target_id, 1, nchar(ovule1$target_id)-2)
row.names(ovule1) <- ovule1$target_id 
#find the expression value from ovule1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sovule1exp <- rep(NA,dim(s_as)[1])
s_as$asovule1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% ovule1$target_id){
    exps<- paste0(ovule1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sovule1exp[i] <- exps}
  if(s_as[i,2]%in% ovule1$target_id){
    expas<- paste0(ovule1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asovule1exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
ovule2<- read.csv("/Users/xung0001/ovule2.tsv",sep="\t")
ovule2 <- as.data.frame(ovule2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(ovule2$target_id)==11, "yes","no")
ovule2 <- ovule2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(ovule2$target_id,nchar(ovule2$target_id),nchar(ovule2$target_id))=="1")
ovule2 <- ovule2[x,]
# remove the .1 from transcript id
ovule2$target_id <- substr(ovule2$target_id, 1, nchar(ovule2$target_id)-2)
row.names(ovule2) <- ovule2$target_id 
#find the expression value from ovule2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sovule2exp <- rep(NA,dim(s_as)[1])
s_as$asovule2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% ovule2$target_id){
    exps<- paste0(ovule2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sovule2exp[i] <- exps}
  if(s_as[i,2]%in% ovule2$target_id){
    expas<- paste0(ovule2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asovule2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
ovule3<- read.csv("/Users/xung0001/ovule3.tsv",sep="\t")
ovule3 <- as.data.frame(ovule3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(ovule3$target_id)==11, "yes","no")
ovule3 <- ovule3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(ovule3$target_id,nchar(ovule3$target_id),nchar(ovule3$target_id))=="1")
ovule3 <- ovule3[x,]
# remove the .1 from transcript id
ovule3$target_id <- substr(ovule3$target_id, 1, nchar(ovule3$target_id)-2)
row.names(ovule3) <- ovule3$target_id 
#find the expression value from ovule3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sovule3exp <- rep(NA,dim(s_as)[1])
s_as$asovule3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% ovule3$target_id){
    exps<- paste0(ovule3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sovule3exp[i] <- exps}
  if(s_as[i,2]%in% ovule3$target_id){
    expas<- paste0(ovule3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asovule3exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
germinating1<- read.csv("/Users/xung0001/germinating1.tsv",sep="\t")
germinating1 <- as.data.frame(germinating1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(germinating1$target_id)==11, "yes","no")
germinating1 <- germinating1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(germinating1$target_id,nchar(germinating1$target_id),nchar(germinating1$target_id))=="1")
germinating1 <- germinating1[x,]
# remove the .1 from transcript id
germinating1$target_id <- substr(germinating1$target_id, 1, nchar(germinating1$target_id)-2)
row.names(germinating1) <- germinating1$target_id 
#find the expression value from germinating1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sgerminating1exp <- rep(NA,dim(s_as)[1])
s_as$asgerminating1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% germinating1$target_id){
    exps<- paste0(germinating1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sgerminating1exp[i] <- exps}
  if(s_as[i,2]%in% germinating1$target_id){
    expas<- paste0(germinating1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asgerminating1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
germinating2<- read.csv("/Users/xung0001/germinating2.tsv",sep="\t")
germinating2 <- as.data.frame(germinating2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(germinating2$target_id)==11, "yes","no")
germinating2 <- germinating2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(germinating2$target_id,nchar(germinating2$target_id),nchar(germinating2$target_id))=="1")
germinating2 <- germinating2[x,]
# remove the .1 from transcript id
germinating2$target_id <- substr(germinating2$target_id, 1, nchar(germinating2$target_id)-2)
row.names(germinating2) <- germinating2$target_id 
#find the expression value from germinating2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sgerminating2exp <- rep(NA,dim(s_as)[1])
s_as$asgerminating2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% germinating2$target_id){
    exps<- paste0(germinating2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sgerminating2exp[i] <- exps}
  if(s_as[i,2]%in% germinating2$target_id){
    expas<- paste0(germinating2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asgerminating2exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
germinating3<- read.csv("/Users/xung0001/germinating3.tsv",sep="\t")
germinating3 <- as.data.frame(germinating3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(germinating3$target_id)==11, "yes","no")
germinating3 <- germinating3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(germinating3$target_id,nchar(germinating3$target_id),nchar(germinating3$target_id))=="1")
germinating3 <- germinating3[x,]
# remove the .1 from transcript id
germinating3$target_id <- substr(germinating3$target_id, 1, nchar(germinating3$target_id)-2)
row.names(germinating3) <- germinating3$target_id 
#find the expression value from germinating3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sgerminating3exp <- rep(NA,dim(s_as)[1])
s_as$asgerminating3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% germinating3$target_id){
    exps<- paste0(germinating3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sgerminating3exp[i] <- exps}
  if(s_as[i,2]%in% germinating3$target_id){
    expas<- paste0(germinating3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asgerminating3exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
germinating4<- read.csv("/Users/xung0001/germinating4.tsv",sep="\t")
germinating4 <- as.data.frame(germinating4)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(germinating4$target_id)==11, "yes","no")
germinating4 <- germinating4[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(germinating4$target_id,nchar(germinating4$target_id),nchar(germinating4$target_id))=="1")
germinating4 <- germinating4[x,]
# remove the .1 from transcript id
germinating4$target_id <- substr(germinating4$target_id, 1, nchar(germinating4$target_id)-2)
row.names(germinating4) <- germinating4$target_id 
#find the expression value from germinating4 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sgerminating4exp <- rep(NA,dim(s_as)[1])
s_as$asgerminating4exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% germinating4$target_id){
    exps<- paste0(germinating4[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sgerminating4exp[i] <- exps}
  if(s_as[i,2]%in% germinating4$target_id){
    expas<- paste0(germinating4[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asgerminating4exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
cotyledon1<- read.csv("/Users/xung0001/cotyledon1.tsv",sep="\t")
cotyledon1 <- as.data.frame(cotyledon1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(cotyledon1$target_id)==11, "yes","no")
cotyledon1 <- cotyledon1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(cotyledon1$target_id,nchar(cotyledon1$target_id),nchar(cotyledon1$target_id))=="1")
cotyledon1 <- cotyledon1[x,]
# remove the .1 from transcript id
cotyledon1$target_id <- substr(cotyledon1$target_id, 1, nchar(cotyledon1$target_id)-2)
row.names(cotyledon1) <- cotyledon1$target_id 
#find the expression value from cotyledon1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$scotyledon1exp <- rep(NA,dim(s_as)[1])
s_as$ascotyledon1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% cotyledon1$target_id){
    exps<- paste0(cotyledon1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$scotyledon1exp[i] <- exps}
  if(s_as[i,2]%in% cotyledon1$target_id){
    expas<- paste0(cotyledon1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ascotyledon1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
cotyledon2<- read.csv("/Users/xung0001/cotyledon2.tsv",sep="\t")
cotyledon2 <- as.data.frame(cotyledon2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(cotyledon2$target_id)==11, "yes","no")
cotyledon2 <- cotyledon2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(cotyledon2$target_id,nchar(cotyledon2$target_id),nchar(cotyledon2$target_id))=="1")
cotyledon2 <- cotyledon2[x,]
# remove the .1 from transcript id
cotyledon2$target_id <- substr(cotyledon2$target_id, 1, nchar(cotyledon2$target_id)-2)
row.names(cotyledon2) <- cotyledon2$target_id 
#find the expression value from cotyledon2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$scotyledon2exp <- rep(NA,dim(s_as)[1])
s_as$ascotyledon2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% cotyledon2$target_id){
    exps<- paste0(cotyledon2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$scotyledon2exp[i] <- exps}
  if(s_as[i,2]%in% cotyledon2$target_id){
    expas<- paste0(cotyledon2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ascotyledon2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
cotyledon3<- read.csv("/Users/xung0001/cotyledon3.tsv",sep="\t")
cotyledon3 <- as.data.frame(cotyledon3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(cotyledon3$target_id)==11, "yes","no")
cotyledon3 <- cotyledon3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(cotyledon3$target_id,nchar(cotyledon3$target_id),nchar(cotyledon3$target_id))=="1")
cotyledon3 <- cotyledon3[x,]
# remove the .1 from transcript id
cotyledon3$target_id <- substr(cotyledon3$target_id, 1, nchar(cotyledon3$target_id)-2)
row.names(cotyledon3) <- cotyledon3$target_id 
#find the expression value from cotyledon3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$scotyledon3exp <- rep(NA,dim(s_as)[1])
s_as$ascotyledon3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% cotyledon3$target_id){
    exps<- paste0(cotyledon3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$scotyledon3exp[i] <- exps}
  if(s_as[i,2]%in% cotyledon3$target_id){
    expas<- paste0(cotyledon3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ascotyledon3exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
cotyledon4<- read.csv("/Users/xung0001/cotyledon4.tsv",sep="\t")
cotyledon4 <- as.data.frame(cotyledon4)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(cotyledon4$target_id)==11, "yes","no")
cotyledon4 <- cotyledon4[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(cotyledon4$target_id,nchar(cotyledon4$target_id),nchar(cotyledon4$target_id))=="1")
cotyledon4 <- cotyledon4[x,]
# remove the .1 from transcript id
cotyledon4$target_id <- substr(cotyledon4$target_id, 1, nchar(cotyledon4$target_id)-2)
row.names(cotyledon4) <- cotyledon4$target_id 
#find the expression value from cotyledon4 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$scotyledon4exp <- rep(NA,dim(s_as)[1])
s_as$ascotyledon4exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% cotyledon4$target_id){
    exps<- paste0(cotyledon4[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$scotyledon4exp[i] <- exps}
  if(s_as[i,2]%in% cotyledon4$target_id){
    expas<- paste0(cotyledon4[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ascotyledon4exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
hypocotyl1<- read.csv("/Users/xung0001/hypocotyl1.tsv",sep="\t")
hypocotyl1 <- as.data.frame(hypocotyl1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(hypocotyl1$target_id)==11, "yes","no")
hypocotyl1 <- hypocotyl1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(hypocotyl1$target_id,nchar(hypocotyl1$target_id),nchar(hypocotyl1$target_id))=="1")
hypocotyl1 <- hypocotyl1[x,]
# remove the .1 from transcript id
hypocotyl1$target_id <- substr(hypocotyl1$target_id, 1, nchar(hypocotyl1$target_id)-2)
row.names(hypocotyl1) <- hypocotyl1$target_id 
#find the expression value from hypocotyl1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$shypocotyl1exp <- rep(NA,dim(s_as)[1])
s_as$ashypocotyl1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% hypocotyl1$target_id){
    exps<- paste0(hypocotyl1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$shypocotyl1exp[i] <- exps}
  if(s_as[i,2]%in% hypocotyl1$target_id){
    expas<- paste0(hypocotyl1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ashypocotyl1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
hypocotyl2<- read.csv("/Users/xung0001/hypocotyl2.tsv",sep="\t")
hypocotyl2 <- as.data.frame(hypocotyl2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(hypocotyl2$target_id)==11, "yes","no")
hypocotyl2 <- hypocotyl2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(hypocotyl2$target_id,nchar(hypocotyl2$target_id),nchar(hypocotyl2$target_id))=="1")
hypocotyl2 <- hypocotyl2[x,]
# remove the .1 from transcript id
hypocotyl2$target_id <- substr(hypocotyl2$target_id, 1, nchar(hypocotyl2$target_id)-2)
row.names(hypocotyl2) <- hypocotyl2$target_id 
#find the expression value from hypocotyl2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$shypocotyl2exp <- rep(NA,dim(s_as)[1])
s_as$ashypocotyl2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% hypocotyl2$target_id){
    exps<- paste0(hypocotyl2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$shypocotyl2exp[i] <- exps}
  if(s_as[i,2]%in% hypocotyl2$target_id){
    expas<- paste0(hypocotyl2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$ashypocotyl2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
rosleaf1<- read.csv("/Users/xung0001/rosleaf1.tsv",sep="\t")
rosleaf1 <- as.data.frame(rosleaf1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(rosleaf1$target_id)==11, "yes","no")
rosleaf1 <- rosleaf1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(rosleaf1$target_id,nchar(rosleaf1$target_id),nchar(rosleaf1$target_id))=="1")
rosleaf1 <- rosleaf1[x,]
# remove the .1 from transcript id
rosleaf1$target_id <- substr(rosleaf1$target_id, 1, nchar(rosleaf1$target_id)-2)
row.names(rosleaf1) <- rosleaf1$target_id 
#find the expression value from rosleaf1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$srosleaf1exp <- rep(NA,dim(s_as)[1])
s_as$asrosleaf1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% rosleaf1$target_id){
    exps<- paste0(rosleaf1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$srosleaf1exp[i] <- exps}
  if(s_as[i,2]%in% rosleaf1$target_id){
    expas<- paste0(rosleaf1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asrosleaf1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
rosleaf2<- read.csv("/Users/xung0001/rosleaf2.tsv",sep="\t")
rosleaf2 <- as.data.frame(rosleaf2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(rosleaf2$target_id)==11, "yes","no")
rosleaf2 <- rosleaf2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(rosleaf2$target_id,nchar(rosleaf2$target_id),nchar(rosleaf2$target_id))=="1")
rosleaf2 <- rosleaf2[x,]
# remove the .1 from transcript id
rosleaf2$target_id <- substr(rosleaf2$target_id, 1, nchar(rosleaf2$target_id)-2)
row.names(rosleaf2) <- rosleaf2$target_id 
#find the expression value from rosleaf2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$srosleaf2exp <- rep(NA,dim(s_as)[1])
s_as$asrosleaf2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% rosleaf2$target_id){
    exps<- paste0(rosleaf2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$srosleaf2exp[i] <- exps}
  if(s_as[i,2]%in% rosleaf2$target_id){
    expas<- paste0(rosleaf2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asrosleaf2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
rosleaf3<- read.csv("/Users/xung0001/rosleaf3.tsv",sep="\t")
rosleaf3 <- as.data.frame(rosleaf3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(rosleaf3$target_id)==11, "yes","no")
rosleaf3 <- rosleaf3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(rosleaf3$target_id,nchar(rosleaf3$target_id),nchar(rosleaf3$target_id))=="1")
rosleaf3 <- rosleaf3[x,]
# remove the .1 from transcript id
rosleaf3$target_id <- substr(rosleaf3$target_id, 1, nchar(rosleaf3$target_id)-2)
row.names(rosleaf3) <- rosleaf3$target_id 
#find the expression value from rosleaf3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$srosleaf3exp <- rep(NA,dim(s_as)[1])
s_as$asrosleaf3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% rosleaf3$target_id){
    exps<- paste0(rosleaf3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$srosleaf3exp[i] <- exps}
  if(s_as[i,2]%in% rosleaf3$target_id){
    expas<- paste0(rosleaf3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asrosleaf3exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
rosleaf4<- read.csv("/Users/xung0001/rosleaf4.tsv",sep="\t")
rosleaf4 <- as.data.frame(rosleaf4)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(rosleaf4$target_id)==11, "yes","no")
rosleaf4 <- rosleaf4[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(rosleaf4$target_id,nchar(rosleaf4$target_id),nchar(rosleaf4$target_id))=="1")
rosleaf4 <- rosleaf4[x,]
# remove the .1 from transcript id
rosleaf4$target_id <- substr(rosleaf4$target_id, 1, nchar(rosleaf4$target_id)-2)
row.names(rosleaf4) <- rosleaf4$target_id 
#find the expression value from rosleaf4 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$srosleaf4exp <- rep(NA,dim(s_as)[1])
s_as$asrosleaf4exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% rosleaf4$target_id){
    exps<- paste0(rosleaf4[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$srosleaf4exp[i] <- exps}
  if(s_as[i,2]%in% rosleaf4$target_id){
    expas<- paste0(rosleaf4[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asrosleaf4exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
roottip_rep1<- read.csv("/Users/xung0001/roottip_rep1.tsv",sep="\t")
roottip_rep1 <- as.data.frame(roottip_rep1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(roottip_rep1$target_id)==11, "yes","no")
roottip_rep1 <- roottip_rep1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(roottip_rep1$target_id,nchar(roottip_rep1$target_id),nchar(roottip_rep1$target_id))=="1")
roottip_rep1 <- roottip_rep1[x,]
# remove the .1 from transcript id
roottip_rep1$target_id <- substr(roottip_rep1$target_id, 1, nchar(roottip_rep1$target_id)-2)
row.names(roottip_rep1) <- roottip_rep1$target_id 
#find the expression value from roottip_rep1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sroottip_rep1exp <- rep(NA,dim(s_as)[1])
s_as$asroottip_rep1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% roottip_rep1$target_id){
    exps<- paste0(roottip_rep1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sroottip_rep1exp[i] <- exps}
  if(s_as[i,2]%in% roottip_rep1$target_id){
    expas<- paste0(roottip_rep1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asroottip_rep1exp[i] <- expas}
}
# loading expression profile generated by from Kallisto method
roottip_rep2<- read.csv("/Users/xung0001/roottip_rep2.tsv",sep="\t")
roottip_rep2 <- as.data.frame(roottip_rep2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(roottip_rep2$target_id)==11, "yes","no")
roottip_rep2 <- roottip_rep2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(roottip_rep2$target_id,nchar(roottip_rep2$target_id),nchar(roottip_rep2$target_id))=="1")
roottip_rep2 <- roottip_rep2[x,]
# remove the .1 from transcript id
roottip_rep2$target_id <- substr(roottip_rep2$target_id, 1, nchar(roottip_rep2$target_id)-2)
row.names(roottip_rep2) <- roottip_rep2$target_id 
#find the expression value from roottip_rep2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sroottip_rep2exp <- rep(NA,dim(s_as)[1])
s_as$asroottip_rep2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% roottip_rep2$target_id){
    exps<- paste0(roottip_rep2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sroottip_rep2exp[i] <- exps}
  if(s_as[i,2]%in% roottip_rep2$target_id){
    expas<- paste0(roottip_rep2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asroottip_rep2exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
im1<- read.csv("/Users/xung0001/im1.tsv",sep="\t")
im1 <- as.data.frame(im1)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(im1$target_id)==11, "yes","no")
im1 <- im1[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(im1$target_id,nchar(im1$target_id),nchar(im1$target_id))=="1")
im1 <- im1[x,]
# remove the .1 from transcript id
im1$target_id <- substr(im1$target_id, 1, nchar(im1$target_id)-2)
row.names(im1) <- im1$target_id 
#find the expression value from im1 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sim1exp <- rep(NA,dim(s_as)[1])
s_as$asim1exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% im1$target_id){
    exps<- paste0(im1[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sim1exp[i] <- exps}
  if(s_as[i,2]%in% im1$target_id){
    expas<- paste0(im1[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asim1exp[i] <- expas}
}

# loading expression profile generated by from Kallisto method
im2<- read.csv("/Users/xung0001/im2.tsv",sep="\t")
im2 <- as.data.frame(im2)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(im2$target_id)==11, "yes","no")
im2 <- im2[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(im2$target_id,nchar(im2$target_id),nchar(im2$target_id))=="1")
im2 <- im2[x,]
# remove the .1 from transcript id
im2$target_id <- substr(im2$target_id, 1, nchar(im2$target_id)-2)
row.names(im2) <- im2$target_id 
#find the expression value from im2 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sim2exp <- rep(NA,dim(s_as)[1])
s_as$asim2exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% im2$target_id){
    exps<- paste0(im2[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sim2exp[i] <- exps}
  if(s_as[i,2]%in% im2$target_id){
    expas<- paste0(im2[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asim2exp[i] <- expas}
}


# loading expression profile generated by from Kallisto method
im3<- read.csv("/Users/xung0001/im3.tsv",sep="\t")
im3 <- as.data.frame(im3)
# keep the transcripts with id length less than 11 character
j <-  ifelse(nchar(im3$target_id)==11, "yes","no")
im3 <- im3[j=="yes",]
# only keep the .1 transcript in the expression file 
x <- which(substr(im3$target_id,nchar(im3$target_id),nchar(im3$target_id))=="1")
im3 <- im3[x,]
# remove the .1 from transcript id
im3$target_id <- substr(im3$target_id, 1, nchar(im3$target_id)-2)
row.names(im3) <- im3$target_id 
#find the expression value from im3 dataset for s_as
s_as <- as.data.frame(s_as)
s_as$sim3exp <- rep(NA,dim(s_as)[1])
s_as$asim3exp <- rep(NA,dim(s_as)[1])
for(i in 1:dim(s_as)[1])
{  print(i)
  exps <- c()
  expas <- c()
  if(s_as[i,1]%in% im3$target_id){
    exps<- paste0(im3[s_as[i,1],]$tpm," ",exps )}
  if(length(exps>0)){s_as$sim3exp[i] <- exps}
  if(s_as[i,2]%in% im3$target_id){
    expas<- paste0(im3[s_as[i,2],]$tpm," ",expas )}
  if(length(expas>0)){s_as$asim3exp[i] <- expas}
}
#two times genermianting seeds
s_as <- s_as[,-c(35:38)]
write.table(s_as, file="sense_antisense_overlappingpair_expression_tissues_allrep.txt", sep="\t", quote=F, row.names=F, col.names=T)
s_as <- read.table("sense_antisense_overlappingpair_expression_tissues_allrep.txt",header = T)
# use mean value to generate heatmap
#select the columns with a specific character
#I have NA in my dataset
s_as[is.na(s_as)] <- 0
c <- which(rowSums(s_as[,-c(1:2)])==0)
s_as <-s_as[-c,]
#require dplyr package
sense <- select(.data = s_as,starts_with("s"))
antisense <- select(.data = s_as,starts_with("as"))
sense$sid <- s_as$V1
sense$asid <- s_as$V2
antisense$sid <- s_as$V1
antisense$asid <- s_as$V2

s_as_f <- data.frame(s_as[,1:2], rowMeans(sense[,1:3]), rowMeans(antisense[,1:3]),rowMeans(sense[,4:6]), rowMeans(antisense[,4:6]),rowMeans(sense[,7:9]), rowMeans(antisense[,7:9]),
                     rowMeans(sense[,10:12]), rowMeans(antisense[,10:12]),rowMeans(sense[,13:16]), rowMeans(antisense[,13:16]),
                     rowMeans(sense[,17:19]), rowMeans(antisense[,17:19]),rowMeans(sense[,20:23]), rowMeans(antisense[,20:23]),
                     rowMeans(sense[,24:27]), rowMeans(antisense[,24:27]), rowMeans(sense[,28:31]), rowMeans(antisense[,28:31]),
                     rowMeans(sense[,32:33]), rowMeans(antisense[,32:33]), rowMeans(sense[,34:36]), rowMeans(antisense[,34:36]))
names(s_as_f) <- c("sid","asid","sroot","asroot","sshoot","asshoot","sinfloresence","asinfloresence","ssilique","assilique","shypocotyl","ashypocotyl","sovule","asovule","sgerminating","asgerminating","scotyledon","ascotyledon","sleaf","asleaf","sroottip","asroottip","sim","asim")
write.table(s_as_f, file="source_heatmap_s_as_overlap.txt", sep="\t", quote=F, row.names=F, col.names=T)
y <- read.table("source_heatmap_s_as_overlap.txt",head=T)
head(y)
y[is.na(y)] <- 0
# delete the rowrums=0
c <- which(rowSums(y[,-c(1:2)])==0)
y <-y[-c,]
## so 2946 pairs have gene expression pattern across tissues
s_as_heatmap <- t(scale(t(y[,-c(1:2)])))
head(s_as_heatmap)
library(pheatmap)
pheatmap(s_as_heatmap, color = colorRampPalette(c('#436eee','white','#EE0000'))(100),cluster_rows = T,cluster_cols = F)
# take the expression pattern of my candidattes across tissues
NRT1.9<-  antisense[ antisense$asid=="AT1G18890",]
CPK10 <- sense[sense$sid=="AT1G18880",]
j <- cbind(t(NRT1.9),t(CPK10))
j
colnames(j) <- c("NRT1.9","CPK10")
j <- j[-(37:38),]
rownames(j)
# remove the first two character of rownames with str_sub
rownames(j) <- str_sub(rownames(j),3,nchar(row.names(j))-3)
j
j<- as.data.frame(j)
j$variable <- row.names(j)
library(reshape2)
jm <-melt(j[,c("variable","NRT1.9","CPK10")],id.vars=1)
names(jm) <- c("tissue","genes","value")
str(jm)
jm$tissue[68:69] <- c("roottip1","roottip2")
jm$tissue[32:33] <- c("roottip1","roottip2")
jm <- jm[!jm$tissue=="hypocotyl4",]
jm$tissue <- str_sub(jm$tissue,1,nchar(jm$tissue)-1) 
jm$tissue <- as.factor(jm$tissue)
str(jm)
jm$value <- as.numeric(jm$value)
jm$value <- log(jm$value+1)
jm <- jm[order(jm$value,decreasing=T),]
library(ggplot2)
library(tidyverse)
dev.off()
p <- ggplot(jm, aes(x=tissue, y=value, col=genes,shape=genes))+geom_point()+stat_smooth()+xlab(NULL)+ ylab("log-transformed TPM")+scale_shape_manual(values = c(17,16))+
  theme(panel.background =element_rect(fill = "white", colour = "black", size = 0.01), axis.text = element_text(size = 10, face = "bold"),
axis.title = element_text(colour="black",size = 10, face = "bold"),
               axis.text.x=element_text(hjust=0.5),
             plot.title = element_text(size = 10, face="italic", hjust = 0.48),
          panel.grid.major =element_line(colour="grey",size=0.5,linetype = 3), panel.grid.minor = element_blank(),
                                       legend.position="right",legend.text = element_text(colour="black",size = 10, face = "italic"))

p

